src/pymovements/dataset/_utils/_yaml.py

```diff
@@ -92,3 +96,32 @@ def type_constructor(
         ) from exc
     except ValueError as exc:
         raise ValueError(f'Unknown {node=}') from exc
+
+
+def substitute_types(data: Any) -> Any:
```

Comment:
I moved this at it makes more sense here, where the `type_constructor` is also impelemented